tests/test-mongoc-command-monitoring.c

```diff
@@ -1745,6 +1745,144 @@ test_killcursors_deprecated (void)
 }
 
 
+static void
+command_failed_reply_command_started_cb (const mongoc_apm_command_started_t *event)
+{
+   cmd_test_t *test;
+
+   if (!strcmp (mongoc_apm_command_started_get_command_name (event),
+                "endSessions")) {
+      /* the test is ending */
+      return;
+   }
+
+   test = (cmd_test_t *) mongoc_apm_command_started_get_context (event);
+   test->started_calls++;
+   bson_destroy (&test->cmd);
+   bson_strncpy (test->db,
+                 mongoc_apm_command_started_get_database_name (event),
+                 sizeof (test->db));
+   bson_copy_to (mongoc_apm_command_started_get_command (event), &test->cmd);
+   bson_strncpy (test->cmd_name,
+                 mongoc_apm_command_started_get_command_name (event),
+                 sizeof (test->cmd_name));
+}
+
+
+static void
+command_failed_reply_command_failed_cb (const mongoc_apm_command_failed_t *event)
+{
+   cmd_test_t *test;
+
+   test = (cmd_test_t *) mongoc_apm_command_failed_get_context (event);
+   test->failed_calls++;
+   bson_destroy (&test->cmd);
+
+   ASSERT_CMPSTR (test->cmd_name,
+                  mongoc_apm_command_failed_get_command_name (event));
+
+   bson_copy_to (mongoc_apm_command_failed_get_reply (event), &test->cmd);
```

Comment:
it's confusing to copy "reply" to "cmd", i think it's better to *not* reuse the cmd_test_t struct in this test and instead make a new struct for the purpose of testing command-failed events' replies